crates/typst-syntax/src/lexer.rs

```diff
@@ -127,24 +191,43 @@ impl Lexer<'_> {
             Some(c) => match self.mode {
                 LexMode::Markup => self.markup(start, c),
                 LexMode::Math => self.math(start, c),
-                LexMode::Code => self.code(start, c),
-                LexMode::Raw => unreachable!(),
+                LexMode::Code(mode) => match mode {
+                    // Code mode might produce false `SyntaxKind::End` tokens on newlines.
```

Comment:
The "might" here is a bit confusing. It sounds like that's maybe happening somewhere else and not definitely right here.

Reply:
How about: "Maybe produce false `SyntaxKind::End` tokens on newlines." ?